libs/pika/async_mpi/include/pika/async_mpi/trigger_mpi.hpp

```diff
@@ -186,6 +177,46 @@ namespace pika::mpi::experimental::detail {
                                     r.op_state.status, PIKA_MOVE(r.op_state.receiver));
                                 break;
                             }
+                            case handler_method::new_task:
+                            {
+                                // The callback will call set_value/set_error inside a new task
+                                // and execution will continue on that thread
+                                add_new_task_request_callback(
+                                    r.op_state.request, p, PIKA_MOVE(r.op_state.receiver));
+                                break;
+                            }
+                            case handler_method::continuation:
+                            {
+                                // The callback will call set_value/set_error
+                                // execution will continue on the callback thread
+                                add_continuation_request_callback<>(r.op_state.request, r.op_state);
+                                break;
+                            }
+                            case handler_method::mpix_continuation:
+                            {
+                                PIKA_DETAIL_DP(mpi_tran<1>,
+                                    debug(str<>("MPIX"), "register_mpix_continuation",
+                                        ptr(r.op_state.request), ptr(r.op_state.request)));
+
+                                MPIX_Continue_cb_function* func = &mpix_callback<operation_state>;
+                                register_mpix_continuation(&r.op_state.request, func, &r.op_state);
+                                {
+                                    PIKA_DETAIL_DP(mpi_tran<1>,
+                                        debug(str<>("MPIX"), "waiting", ptr(r.op_state.request)));
+                                    std::unique_lock l{r.op_state.mutex};
+                                    r.op_state.cond_var.wait(
+                                        l, [&]() { return r.op_state.completed; });
+                                }
+
+                                PIKA_DETAIL_DP(mpi_tran<1>,
+                                    debug(str<>("MPIX"), "woken", ptr(r.op_state.request)));
+
+                                // call set_value/set_error depending on mpi return status
+                                set_value_error_helper(
+                                    /*r.op_state.status*/ MPI_SUCCESS,
```

Comment:
Always success?

Reply:
changed to use the rc return from the callback